diff --git a/db/internal_stats.h b/db/internal_stats.h
index ca07305..d54ff18 100644
--- a/db/internal_stats.h
+++ b/db/internal_stats.h
@@ -170,6 +170,8 @@ class InternalStats {
     // Number of compactions done per CompactionReason
     int counts[static_cast<int>(CompactionReason::kNumOfReasons)];
 
+    CompactionStats& operator=(CompactionStats const&) = default;
+
     explicit CompactionStats()
         : micros(0),
           bytes_read_non_output_levels(0),
diff --git a/db/version_edit.h b/db/version_edit.h
index 808be58..08c2262 100644
--- a/db/version_edit.h
+++ b/db/version_edit.h
@@ -52,14 +52,9 @@ struct FileDescriptor {
         smallest_seqno(_smallest_seqno),
         largest_seqno(_largest_seqno) {}
 
-  FileDescriptor& operator=(const FileDescriptor& fd) {
-    table_reader = fd.table_reader;
-    packed_number_and_path_id = fd.packed_number_and_path_id;
-    file_size = fd.file_size;
-    smallest_seqno = fd.smallest_seqno;
-    largest_seqno = fd.largest_seqno;
-    return *this;
-  }
+  FileDescriptor(FileDescriptor const&) = default;
+
+  FileDescriptor& operator=(FileDescriptor const&) = default;
 
   uint64_t GetNumber() const {
     return packed_number_and_path_id & kFileNumberMask;
diff --git a/utilities/persistent_cache/persistent_cache_util.h b/utilities/persistent_cache/persistent_cache_util.h
index 214bb58..254c038 100644
--- a/utilities/persistent_cache/persistent_cache_util.h
+++ b/utilities/persistent_cache/persistent_cache_util.h
@@ -48,7 +48,7 @@ class BoundedQueue {
     T t = std::move(q_.front());
     size_ -= t.Size();
     q_.pop_front();
-    return std::move(t);
+    return t;
   }
 
   size_t Size() const {
